### PR TITLE
ServerUploader: detecting invalid values (NaN/inf) in the vector and sending error as low frequency event (once every 5 minutes). #801

### PR DESCRIPTION
**Pull request by LarsWithCA**
Example IO.conf:
image

B.log:
image

Events:
image

### CODE REVIEW

File: CA_DataUploaderLib/ServerUploader.cs [hunk 1]
```diff
@@ -94,19 +109,30 @@ private void SendVector(DataVector vector)
...
+            if (!string.IsNullOrWhiteSpace(invalidValueMessage))
+                SendLowFrequencyEvent(new EventFiredArgs("Warning - invalid values detected: " + invalidValueMessage[2..], EventType.Log, DateTime.UtcNow));
 
-            DataVector WithOnlyUploadFields(DataVector fullVector)
+            DataVector FilterOnlyUploadFieldsAndCheckInvalidValues(DataVector fullVector, out string invalidValueMessage)
             {
                 var fullVectorData = fullVector.Data;
                 var uploadData = new double[uploadDesc.Length];
                 int j = 0;
+                StringBuilder invalidValues = new();
```
Comment (freddyrios) [inline CA_DataUploaderLib/ServerUploader.cs:123]: This allocates memory on every vector even when not tripping

File: CA_DataUploaderLib/ServerUploader.cs [hunk 2]
```diff
@@ -37,6 +37,8 @@ public sealed class ServerUploader : ISubsystemWithVectorData
         private readonly IIOconf _ioconf;
         private readonly CommandHandler _cmd;
         private readonly TaskCompletionSource<PlotConnection> _connectionEstablishedSource = new();
+        private readonly Stopwatch _timeSinceLastLowFrequencyEvent = new();
```
Comment (freddyrios) [inline CA_DataUploaderLib/ServerUploader.cs:40]: This watch should probably be specific to the special double values detection.

**Review by freddyrios — APPROVED** [review]

File: CA_DataUploaderLib/ServerUploader.cs (hunk 1 above)
Comment (LarsWithCA) [reply CA_DataUploaderLib/ServerUploader.cs]: fixed

File: CA_DataUploaderLib/ServerUploader.cs (hunk 2 above)
Comment (LarsWithCA) [reply CA_DataUploaderLib/ServerUploader.cs]: done

**Pull request merged**